packages/flutter/lib/src/rendering/text_boundary.dart

```diff
@@ -0,0 +1,174 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:characters/characters.dart' show CharacterRange;
+import 'package:flutter/services.dart';
+
+/// An interface for retrieving the logical text boundary (left-closed-right-open)
+/// at a given location in a document.
+///
+/// Depending on the implementation of the [TextBoundary], the input
+/// [TextPosition] can either point to a code unit, or a position between 2 code
```

Comment:
nit: "code unit" -> "UTF-16 code unit"

Reply:
Actually I think as a public API it will be easier to interpret all `TextPosition`s as caret locations (since it has an affinity).